go-controller/pkg/ovn/egressip.go

```diff
@@ -206,6 +219,165 @@ func (oc *Controller) syncEgressIPs(eIPs []interface{}) {
 			klog.Errorf("Unable to delete legacy logical router policy: %s, stderr: %s, err: %v", policyID, stderr, err)
 		}
 	}
+
+	// This part will take of syncing stale data which we might have in OVN if
+	// there's no ovnkube-master running for a while, while there are changes to
+	// pods/egress IPs.
+	// It will sync:
+	// - Egress IPs which have been deleted while ovnkube-master was down
+	// - pods/namespaces which have stopped matching on egress IPs while
+	//   ovnkube-master was down
+	if egressIPToPodIPCache, err := oc.generatePodIPCacheForEgressIP(eIPs); err == nil {
+		oc.syncStaleEgressReroutePolicy(egressIPToPodIPCache)
+		oc.syncStaleNATRules(egressIPToPodIPCache)
+	}
+}
+
+func (oc *Controller) syncStaleEgressReroutePolicy(egressIPToPodIPCache map[string]sets.String) {
```

Comment:
general comment: is it better to read these fields as json and read into fields of a struct?

Reply:
It could be I guess, I don't have a strong opinion. It might be easier to read. I can update that

Reply:
we have a mix of json, csv, whitespace parsing (gross) all around the code. Since we are going to go bindings we will get rid of all of this. I'm ok with using csv or json here. @abhat do you strongly prefer json, or can we merge the PR as is?

Reply:
Aniket is OK with it (slack discussion)